packages/swap/src/__tests__/amm.test.ts

```diff
@@ -0,0 +1,70 @@
+import BigNumber from 'bignumber.js';
+import { Side, swap } from '@/amm-addon';
+import { InternalAsset, assetConstants } from '@/shared/enums';
+
+const toAtomicUnits = (amount: number, asset: InternalAsset) =>
+  BigInt(new BigNumber(amount).shiftedBy(assetConstants[asset].decimals).toFixed(0));
+
+describe(swap, () => {
+  it('gets the price for expected price (USDC => USDT)', async () => {
+    const { remainingAmount, swappedAmount } = await swap({
```

Comment:
not quite sure what this swap function is doing. is this against a pool?

Reply:
yeah it runs it through the Rust AMM code in the native addon